lib/rubocop/ast/node/self_node.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module AST
+    # A node extension for `self` nodes. This will be used in place of a
+    # plain node when the builder constructs the AST, making its methods
+    # available to all `self` nodes within RuboCop.
+    class SelfNode < Node
+      include MethodIdentifierPredicates
+
+      # Always return `false` because `self` cannot have arguments.
+      def arguments?
```

Comment:
Can you point me to the code that calls `#arguments?` on a `self` node? I think the call site probably should filter the relevant nodes instead. I can have a look. 🙂

Reply:
`SelfNode#arguments?` is called in the following code.
https://github.com/rubocop-hq/rubocop/blob/v0.80.0/lib/rubocop/cop/style/one_line_conditional.rb#L96

I was worried about choosing between implementing `SelfNode#arguments?` or guarding the above with `resopnd_to(:arguments?)`.

```diff
  def keyword_with_changed_precedence?(node)
    return false unless node.keyword?
    return true if node.prefix_not?

-   node.arguments? && !node.parenthesized_call?
+   node.respond_to?(:arguments?) && node.arguments? && !node.parenthesized_call?
  end
```

Reply:
@koic Let me have a look. 🙂 Ideally we shouldn't do any of those. We should make sure incompatible nodes don't reach this code.